app/src/main/java/com/example/bigowlapp/activity/ViewAnotherUserActivity.java

```diff
@@ -1,112 +1,133 @@
-                if (task.isSuccessful()) {
-                    Toast.makeText(ViewAnotherUserActivity.this,
-                            "You have sent a request to supervise ",
-                            Toast.LENGTH_SHORT).show();
-                    supRequestStatus = "sent_pending";
-                    supRequestBtn.setText(supBtn3);
-                } else {
-                    Toast.makeText(ViewAnotherUserActivity.this,
-                            "" + Objects.requireNonNull(task.getException()).toString(),
-                            Toast.LENGTH_SHORT).show();
-                }
-            });
+    private void DoRequest() {
+        SupervisionRequest supervisionRequest = new SupervisionRequest();
+        supervisionRequest.setReceiverUId(otherUserID);
+        supervisionRequest.setSenderUId(currentUserID);
+        supervisionRequest.setResponse(SupervisionRequest.Response.NEUTRAL);
+        supervisionRequest.setGroupUId(""); // TODO think about creating and setting group IDs
+        supervisionRequest.setTimeSent(Timestamp.now());
+        supervisionRequest.setTime(Timestamp.now());
+
+        if (!aRequestAlready) {
+            notificationRepository.addDocument(supervisionRequest);
+
+        } else if (shouldCancelRequest) {
+            notificationRepository.removeDocument(requestUID);
         }
-        if (supRequestStatus.equals("sent_pending") || supRequestStatus.equals("sent_declined")) {
-            supRequestRef.child(mUser.getUid()).child(otherUserID)
-                    .removeValue()
-                    .addOnCompleteListener((OnCompleteListener<Void>) task -> {
-                        if (task.isSuccessful()) {
-                            Toast.makeText(ViewAnotherUserActivity.this,
-                                    "You have canceled request to supervise ",
-                                    Toast.LENGTH_SHORT).show();
-                            supRequestStatus = "none";
-                            supRequestBtn.setText(supBtn1);
-                        } else {
-                            Toast.makeText(ViewAnotherUserActivity.this,
-                                    "" + Objects.requireNonNull(task.getException()).toString(),
-                                    Toast.LENGTH_SHORT).show();
+
+        observeRequests();
+    }
+
+    private void observeRequests() {
+        aRequestAlready = false; // A false value allows for search for an existing request with
+                                // in repository until one is found
+        supRequestBtn.setText(supBtnSend); // Default setText
+        noteTv2.setText(noRequest);
+        LiveData<List<SupervisionRequest>> senderRequestsData = notificationRepository.getListOfSupervisionRequestByAttribute("senderUId", currentUserID, SupervisionRequest.class);
+        senderRequestsData.observe(this, senderRequests -> {
+            if (senderRequests != null) {
```

Comment:
flip this if statement to reduce nesting. Example below...

```java
if (senderRequests == null) {
    return;
}
// Do rest of logic here
```

Reply:
Since it is a void method, the return will be unnecessary and the if statement will be empty

Reply:
You can use `return` in void methods to exit early, such as for failure scenarios, hence...

```java
public void foo() {
    if (x == null) {
        return;
    }
    // Do rest of logic here
}
```

is exactly the same as...

```java
public void foo() {
    if (x != null) {
        // Do rest of logic here
    }
}
```

I recommended it (the first way) since SonarCloud suggested there was too much nesting (which I agree with), and that reduces one nesting level and is also more readable.

Reply:
ok done